src/ng/q.js

```diff
@@ -276,7 +315,7 @@ function qFactory(nextTick, exceptionHandler) {
 
       this.$$state.pending = this.$$state.pending || [];
       this.$$state.pending.push([result, onFulfilled, onRejected, progressBack]);
-      if (this.$$state.status > 0) scheduleProcessQueue(this.$$state);
+      if (this.$$state.status > 0) scheduleProcessQueue(result.promise, this.$$state);
```

Comment:
`$$state.pending` is an array that has all the promises that should be fulfill once this promise is fulfill. The check `if (this.$$state.status > 0)` checks if this promise is fulfill. At `scheduleProcessQueue` we check if the given `$$state` was scheduled before, and if that is the case, then do nothing.

This is what the comment https://github.com/angular/angular.js/pull/11159/files#r26261523 is talking about.

Eg

```js
var parent = $q.when('XXX'); // parent is fulfill
var child1 = parent.then(...); // given that parent is fulfill, parent.$$state is scheduled (but not called)
var child2 = parent.then(...); // parent.$$state is already scheduled, then it is not scheduled again
```

So, this implies that the current cases will have some discrepancies
```js
// Case 1, there is a previously schedule promise resolution
var parent = $qRaf.when('XXX'); // parent is fulfill
var child1 = parent.then(...); // parent.$$state is scheduled (but not called)
$$rAF(function() {
  var child2 = parent.then(...); // parent.$$state is already scheduled, so this is going to be resolve with `$$rAF(callback)`
});
```


```js
// Case 2, there is no previously schedule promise resolution
var parent = $qRaf.when('XXX'); // parent is fulfill
$$rAF(function() {
  var child2 = parent.then(...); // parent.$$state is not scheduled, so this is going to be resolve synchronously
});
```


In the case I ma missing something, please let me know